Lib/Cache/Cache.php

```diff
@@ -0,0 +1,638 @@
+<?php
+/**
+ * The following is a modified version of https://github.com/cakephp/cakephp/blob/2.10.22/lib/Cake/Cache/Cache.php
+ * The modifications are limited to the addition of the the `$parentKey` parameter to the `write` method.
+ */
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @package       Cake.Cache
+ * @since         CakePHP(tm) v 1.2.0.4933
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+
+App::uses('Inflector', 'Utility');
+App::uses('CacheEngine', 'Cache');
+
+/**
+ * Cache provides a consistent interface to Caching in your application. It allows you
+ * to use several different Cache engines, without coupling your application to a specific
+ * implementation. It also allows you to change out cache storage or configuration without effecting
+ * the rest of your application.
+ *
+ * You can configure Cache engines in your application's `bootstrap.php` file. A sample configuration would
+ * be
+ *
+ * ```
+ *	Cache::config('shared', array(
+ *		'engine' => 'Apc',
+ *		'prefix' => 'my_app_'
+ *  ));
+ * ```
+ *
+ * This would configure an APC cache engine to the 'shared' alias. You could then read and write
+ * to that cache alias by using it for the `$config` parameter in the various Cache methods. In
+ * general all Cache operations are supported by all cache engines. However, Cache::increment() and
+ * Cache::decrement() are not supported by File caching.
+ *
+ * @package       Cake.Cache
+ */
+class Cache {
```

Comment:
@jec3 Could you just extend the Cache class so you only have to override the `write()` method instead of having to copy the entire class?

Reply:
> @jec3 Could you just extend the Cache class so you only have to override the `write()` method instead of having to copy the entire class?

@dchancogne I want all of the existing uses of the Cache class to use `write` override. If we extend Cache we'll have to update all existing uses of the Cache class to use our new class. This seems messier than overriding the class entirely.

Reply:
Makes sense and the header comment is clear.